### PR TITLE
Jimmy elif

### PR DESCRIPTION
**Pull request by rxie25**
Major change to if statements to include `elif`.

### CODE REVIEW

**Comment by rxie25** [comment]
@tiancizhong can you take a look at the semant.ml for `if`. Thank you.

**Comment by jzacsh** [comment]
Does this mean we would not support `} else if (expr) {`? _(note: uses our existing keywords)_

**Comment by rxie25** [comment]
We said we will use our existing keywords? I think somehow I missed that. I was just following our LRM. It's shouldn't be hard to change it.

**Comment by rxie25** [comment]
@jzacsh  This is finally done. Take a look.

File: src/ast.ml [hunk 1]
```diff
@@ -63,26 +63,34 @@ let string_of_uop = function
...
+        f ^ "(" ^ String.concat ", " (List.map string_of_expr el) ^ ")"
 
 let rec string_of_stmt = function
     Block(stmts) ->
       "{\n" ^ String.concat "" (List.map string_of_stmt stmts) ^ "}\n"
   | Expr(expr) -> string_of_expr expr ^ ";\n";
-  | If(e, s, Block([])) -> "if (" ^ string_of_expr e ^ "){\n" ^string_of_stmt s^"}"
-  | If(e, s1, s2) ->  "if (" ^ string_of_expr e ^ "){\n" ^
-      string_of_stmt s1 ^ "}else{\n" ^ string_of_stmt s2 ^"}"
+  | If(conds) -> "\n" ^ string_of_first_cond_stmts (List.hd conds) ^ "\n" ^
```
Comment (jzacsh) [inline src/ast.ml]: you fixed the whitespace to be readable up above. can you do the same here for this `If` match block?

File: src/ast.ml [hunk 2]
```diff
@@ -63,26 +63,34 @@ let string_of_uop = function
...
+  (String.concat "\n" (List.map string_of_cond_stmts (List.tl conds)))
   | While(e, s) -> "while (" ^ string_of_expr e ^ ") " ^ string_of_stmt s
 
+  and string_of_first_cond_stmts = function
+    | (None, Block(stmts)) -> "else {\n" ^ (String.concat "\n" (List.map string_of_stmt stmts))
+    | (Some(expr), Block(stmts)) -> "if (" ^ (string_of_expr expr) ^ ")\n {\n" ^ (String.concat "\n" (List.map string_of_stmt stmts)) ^ "}\n"
```
Comment (jzacsh) [inline src/ast.ml]: _(more readability)_ can you add line breaks in your code every so-often after a `^` string concat?

Comment (rxie25) [inline src/ast.ml]: sure

File: src/ast.ml (hunk 1 above)
Comment (rxie25) [inline src/ast.ml]: ok

**Comment by rxie25** [comment]
It's done @jzacsh

**Comment by jzacsh** [comment]
looks good to me!

**Pull request merged**